simulation/simple_sensor_simulator/test/src/sensor_simulation/primitives/test_primitive.cpp

```diff
@@ -0,0 +1,250 @@
+// Copyright 2015 TIER IV, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "test_primitive.hpp"
+
+#include <limits>
+
+#include "../../utils/expect_eq_macros.hpp"
+
+/**
+ * @note Test basic functionality. Test adding to scene correctness with a sample primitive.
+ */
+TEST_F(PrimitiveTest, addToScene_sample)
+{
+  const std::vector<Vertex> expected_vertices = {
+    {0.0f, 1.0f, -1.0f}, {2.0f, 1.0f, -1.0f}, {0.0f, 3.0f, -1.0f}, {2.0f, 3.0f, -1.0f},
+    {0.0f, 1.0f, 1.0f},  {2.0f, 1.0f, 1.0f},  {0.0f, 3.0f, 1.0f},  {2.0f, 3.0f, 1.0f}};
+  const auto expected_triangles = primitive_->getTriangles();
```

Comment:
Please do not use dummy primitive.
The code is written to create a dummy primitive and test it, which is not a test that contributes to improving the quality of simple_sensor_simulator.

Reply:
It's resolved. I used the Box class, but although I added a unit test for 'Test function behavior with vertices and triangles set to only zeros' to the Primitive class, I couldn't set the triangles to zero because they are preset. I set the vertices to zero by setting the box dimensions to 0.